In [ ]:
# Remaining items:
# - Grab code from download notebook
# - Update how we handle paths

# Converting from LSST Skymaps

## Why do we want to convert skymaps in the first place?

~~Short answer: spheres are evil~~

Long answer: we need to convert the [lsst.skymap](https://github.com/lsst/skymap) type skymap that LSST uses if we wish to use it in environments that lack access to the LSST stack.

We *could* try to shim some of the LSST-stack dependencies that lsst.skymaps has, but unfortunately, skymaps use a lot of spherical geometry that would require essentially rewriting a number of other packages from scratch.

See the Skymaps Overview notebook for a little more explanation about why skymaps need these kinds of mathy packages.

## What to do we do to convert the skymap?

### Download the LSST skymap

This next section assumes you have the required access. If not, you don't actually really need to be reading this notebook--we've included an already-converted LSST skymap in this package at: (TODO path or method to get the path)

TODO : copy and paste and revise from "Download..." notebook

### Readin the raw skymap

TODO : change path I guess?

In [4]:
from pathlib import Path

from skymap_convert.utils import load_pickle_skymap

package_root = Path.home() / "skymap-convert"

raw_skymaps_dir = package_root / "tests" / "data" / "raw_skymaps"
raw_skymap_path = raw_skymaps_dir / "skyMap_lsst_cells_v1_skymaps.pickle"

In [5]:
lsst_skymap = load_pickle_skymap(raw_skymap_path)
lsst_skymap

### Using the ConvertedSkymap writer

In [ ]:
from skymap_convert.skymap_writers import ConvertedSkymapWriter

# Create a writer for the LSST skymap
writer = ConvertedSkymapWriter()

# Designate the output path for the converted skymap
converted_skymap_path = package_root / "converted_skymaps" / "lsst_skymap"

# Write the skymap to a file - warning: this may take a few minutes!
writer.write(lsst_skymap, output_path=converted_skymap_path)

Let's take a quick look at the files we created.

In [ ]:
!ls -lh {converted_skymap_path}

total 29M
-rw-r--r-- 1 olynn gu  123 Jul 10 02:05 metadata.yaml
-rw-r--r-- 1 olynn gu  28M Jul 10 02:05 patches.npy.gz
-rw-r--r-- 1 olynn gu 1.2M Jul 10 02:05 tracts.npy


Note that our `patches.npy` has been gzipped--without this compression, the file would be about 1.2 MB, just a little too big to be hosted on GitHub. 

Don't worry; the skymap writer and reader will handle gzipping and un-gzipping under the hood.

## About  

**Author:** Olivia Lynn

**Last updated on:** July 11, 2025